weave-js/src/components/PagePanelComponents/Home/Browse3/inference/util.ts

```diff
@@ -0,0 +1,204 @@
+import logoIBM from './logos/icon-ibm.png';
+import logoMeta from './logos/icon-meta.png';
+import logoMicrosoft from './logos/icon-microsoft.png';
+import logoMistral from './logos/icon-mistral.png';
+import {InferenceContext, InferenceContextType, Model, Source} from './types';
+
+// TODO: Duplicating this from core - move into context?
+export const INFERENCE_PATH = '/inference';
+
+export function urlInference(collectionId?: string, modelId?: string): string {
+  const parts = [INFERENCE_PATH];
+  if (collectionId) {
+    parts.push(collectionId);
+  }
+  if (modelId) {
+    parts.push(modelId);
+  }
+  return parts.join('/');
+}
+
+const sourceNameMap: Record<Source, string> = {
+  BAAI: 'Beijing Academy of Artificial Intelligence (BAAI)',
+  deepseek: 'DeepSeek',
+  'deepseek-ai': 'DeepSeek',
+  'ibm-granite': 'IBM',
+  'meta-llama': 'Meta',
+  microsoft: 'Microsoft',
+  mistral: 'Mistral',
+  mistralai: 'Mistral',
+  openai: 'OpenAI',
+  Qwen: 'Alibaba',
+  xai: 'xAI',
+};
+
+const sourceLogoMap: Record<Source, string> = {
+  BAAI: 'baai',
+  deepseek: 'deepseek',
+  'deepseek-ai': 'deepseek',
+  'ibm-granite': 'ibm',
+  'meta-llama': 'meta',
+  microsoft: 'microsoft',
+  mistral: 'mistral',
+  mistralai: 'mistral',
+  openai: 'openai',
+  Qwen: 'alibaba',
+};
+
+const logoIdToImageMap: Record<string, any> = {
+  alibaba: logoAlibaba,
+  baai: logoBAAI,
+  deepseek: logoDeepSeek,
+  ibm: logoIBM,
+  meta: logoMeta,
+  microsoft: logoMicrosoft,
+  mistral: logoMistral,
+};
+
+export const getModelSource = (model: Model): Source => {
+  return (
+    model.source ??
+    (model.idHuggingFace ? model.idHuggingFace.split('/')[0] : 'unknown')
+  );
+};
+
+export const getSourceLogo = (source: Source) => {
+  const logoId = sourceLogoMap[source];
+  if (!logoId) {
+    return null;
+  }
+  return logoIdToImageMap[logoId] ?? null;
+};
+
+export const getModelSourceName = (model: Model) => {
+  const source = getModelSource(model);
+  return sourceNameMap[source];
+};
+
+export const getModelLogo = (model: Model) => {
+  const source = getModelSource(model);
+  return getSourceLogo(source);
+};
+
+export const getModelLabel = (model: Model) => {
+  return model.label ?? model.id.split('/').pop();
+};
+
+export const getLaunchDateString = (model: Model) => {
+  if (model.launchDate) {
+    const date = new Date(model.launchDate);
+    return date.toLocaleDateString('en-US', {month: 'short', year: 'numeric'});
+  }
+  return '';
+};
+
+const toDollarsPerMillionTokens = (priceCentsPerBillionTokens: number) => {
+  const centsPerDollar = 100;
+  const milPerBil = 1000;
+  return priceCentsPerBillionTokens / (centsPerDollar * milPerBil);
+};
+
+const getRoundedPriceString = (price: number) => {
+  return Number(Math.round(price * 100) / 100)
+    .toFixed(2)
+    .replace(/\.?0+$/, '');
+};
+
+export const getOnePriceString = (priceCentsPerBillionTokens: number) => {
+  return `$${getRoundedPriceString(
```

Comment:
wow maybe the only conceivable time when using the `${}` syntax is confusing (when there is an actual $ char prepended)